### ISSUE TITLE
Diffuse Material and Video Frames

### ISSUE THREAD

**Issue by Steph55** [helix-toolkit/helix-toolkit#1900]
The helix toolkit is surprisingly versatile and powerful for free software, thank you very much!

I'm now trying to feed video frames to a diffuse material's Texture2D. Of course, the texture must be dynamic and the Helix Toolkit does not directly provide this functionality. However, I'm making a custom ITextureResourceManager to achieve this. I still want to cache the other textures as usual, so I basically copy the code from TextureResourceManager for them.

The problem is that line 90 (proxy.Guid = textureModel.Guid;) is not valid in my custom class, since the ShaderResourceViewProxy.Guid is internal set. Would it be a good thing for you to make it public, or should I achieve my goal in a completely different way?

**Comment by holance** [comment]
I can change the Guid to be public.

**Comment by Steph55** [comment]
I finally decided not to do this and concentrate my efforts on other parts of my software. However, I think that I had everything ready for that. I cannot really share the required parts of my software because they span many different objects.

A few ideas for you. First, to display correctly anything else that very small videos, all your frames should be in the GPU right from the beginning (from decompression). I use MediaFoundation for this (with a supplied ID3D11Device), but it is quite complex, even more with WPF (if this is what you're using like me) because MF doesn't work on the STA thread (main thread). I did my own Direct3D swapchain renderer with 5 backbuffers in order to stabilize the videos. I use Helix in the offscreen rendering scheme, in this dedicated thread.

I copy every frame from the Media Foundation IMFSample to an ID3D11Texture2D. However, make sure that you reuse allocated textures, because allocating new ones takes a lot of GPU time!

Good luck!
